### DISCUSSION TITLE
BAM/SAM input option

### DISCUSSION THREAD

**Discussion by apollo994** [ConesaLab/SQANTI3#273] [Ideas] [answered]
Hello SQANTI team,

I've started using SQANTI3 to directly evaluate reads instead of gene models. This is a really helpful feature when comparing two sequencing runs/technologies and you want to avoid the bias from the transcript model construction algorithm. In this context, it would be helpful to have an option to input an alignment file directly. This would skip the time/resource consuming step that, in many cases, has already been done in previous steps.

Thanks a lot for sharing and maintaining this project!

**Comment by carolinamonzo** [comment, marked as answer]
Hi @apollo994! We agree, it's something that we have in mind but haven't had the time to implement it yet. However, you can give a GFF to SQANTI instead of starting from scratch and having to construct transcript models.

What you can do is use the [spliced_bam2gff](https://github.com/nanoporetech/spliced_bam2gff) tool from Nanopore to transform your bam to GFF (careful with the -t option, it is a reported issue in the tool's repository that instead of corresponding to threads, it's a threshold of to classify all deletions larger than the value as introns). And then run SQANTI on the GFF of your reads.

**Reply by carolinamonzo** [reply]
Since this enhancement suggestion is interesting, now moved to Issue #274 for enhancement in followup versions.

**Comment by TianYuan-Liu** [comment]
Hey Fabio,
thanks for the update! I'm currently working on implementing that feature you mentioned. Your input is really helpful in improving SQANTI3. I'll keep you posted on the progress.

**Reply by apollo994** [reply]
Hi Tian, 
thanks for the heads-up. One suggestion I have is to look into parallelisation of the QC step. When I run SQANTI on reads I downsampled my FASTQ to 1M reads to get the result in less than 1h. I tested the `-t` (should only be relevant for minimap) and the `-n` option but I could not see SQANTI taking advantage of it. The last section of the [manual](https://github.com/ConesaLab/SQANTI3/wiki/Running-SQANTI3-Quality-Control#parallelization)  where the relationship between CPUs and chunks is explained is not clear to me. I tried `-t 4 -n 4` in a 4 cores instance on our HPC, hower I could only see 2 cores beign used.

**Reply by TianYuan-Liu** [reply]
Hi Fabio,
Thanks for your suggestion on parallelizing the QC step. It's puzzling that SQANTI didn't fully utilize all 4 cores despite using -t 4 -n 4. This might be due to how SQANTI handles parallel processes or a specific configuration issue. I will look into it and also the relevant issue #243. Meanwhile, experimenting with different -t and -n values may help find a more effective setup on your HPC. Thanks
